tests/integ/test_column_names.py

```diff
@@ -648,49 +763,89 @@ def test_binary_expression(session):
+            else '"((""B"" = \'X\') AND (""A"" = 1 :: INT))"',
+            '"((""B"" = \'X\') OR (""A"" = 1))"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"((""B"" = \'X\') OR (""A"" = 1 :: INT))"',
+            '"BITAND(1, ""A"")"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"BITAND(1 :: INT, ""A"")"',
+            '"BITOR(1, ""A"")"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"BITOR(1 :: INT, ""A"")"',
+            '"BITXOR(1, ""A"")"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"BITXOR(1 :: INT, ""A"")"',
+            '"POWER(""A"", 2)"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"POWER(""A"", 2 :: INT)"',
             '"(""A"" + ""A"")"',
             '"(""A"" - ""A"")"',
             '"(""A"" * ""A"")"',
             '"(""A"" / ""A"")"',
             '"(""A"" % ""A"")"',
-        ]
+        ],
+        [
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            LongType() if not local_testing_mode else BooleanType(),
+            LongType() if not local_testing_mode else BooleanType(),
+            LongType() if not local_testing_mode else BooleanType(),
+            DoubleType(),
+            LongType(),
+            LongType(),
+            LongType(),
+            DecimalType(38, 6),
+            LongType(),
+        ],
+        # Not able to collect result, failed with error "Numeric value 'x' is not recognized"
+        None,
     )
 
 
 def test_cast_nan_column_name(session):
     df1 = session.create_dataframe([["a"]], schema=["a"])
     df2 = df1.select(df1["A"] == math.nan)
-    assert (
-        df2._output[0].name
-        == df2.columns[0]
-        == get_metadata_names(session, df2)[0]
-        == '"(""A"" = \'NAN\' :: FLOAT)"'
+
+    verify_column_result(
+        session,
+        df2,
+        ['"(""A"" = \'NAN\' :: FLOAT)"'],
+        [BooleanType()],
+        # Not able to collect, failed due to "Numeric value 'a' is not recognized"
+        None,
     )
 
 
-def test_inf_column_name(session):
+def test_inf_column_name(session, local_testing_mode):
     df1 = session.create_dataframe([["inf"]], schema=["'INF'"])
     df2 = df1.select(df1["'INF'"] == math.inf)
-    assert (
-        df2._output[0].name
-        == df2.columns[0]
-        == get_metadata_names(session, df2)[0]
-        == '"(""\'INF\'"" = \'INF\' :: FLOAT)"'
+
+    # local testing mode is not handling INF correctly today, skip the result
+    # value check when it is local testing mode
+    expected_rows = None if local_testing_mode else [Row(True)]
+    verify_column_result(
+        session,
+        df2,
+        ['"(""\'INF\'"" = \'INF\' :: FLOAT)"'],
```

Comment:
@sfc-gh-aling @sfc-gh-stan This case seems returning False for local testing mode, is that expected?

Reply:
Can you elaborate please? What's not being handled correctly?

Reply:
Hi @sfc-gh-stan sorry for the late reply. For this test case, it has a dataframe with one single column named 'INF', and have value 'inf', which is  counted as np.inf, and the test is simply checking if df1["'INF'"] == math.inf, which should be True, Snowpark python seems returning the correct result, but with local testing the result seems False.

Reply:
No worries, what is `df1["'INF'"]` returning for local testing then? It should also be `np.inf`.

Reply:
Ok, i think i know what is happening there. when we create a dataframe like this 
```
session.create_dataframe([["inf"]], schema=["'INF'"])
```
the column is actually mapped to string column, in snowflake, when comparing with 'inf' = 'INF'::FLOAT, it was able to do property casting, and return True. However, in local testing i believe it is not able to do such casting, in python, string to numeric value is not castable. That is the behavior difference between python and snowflake

Reply:
This is [coercion](https://docs.snowflake.com/en/sql-reference/data-type-conversion).
How did the test work before then?

Reply:
The test doesn't check any result before, I have updated the whole test suite to check the result type and result in this change, and start seeing  mismatches here. It seems that the local testing is probably lacking some test coverage for special values.

Reply:
I created [SNOW-1524637](https://snowflakecomputing.atlassian.net/browse/SNOW-1524637), please put this ticket number wherever you see this issue happen.

Reply:
added comment with ticket number